tools/src/main/scala/scala/scalanative/codegen/CodeGen.scala

```diff
@@ -43,29 +43,33 @@ object CodeGen {
 
   /** Generate code for given assembly. */
   private def emit(config: build.Config, assembly: Seq[Defn])(
-      implicit meta: Metadata): Unit =
+      implicit meta: Metadata): Seq[Path] =
     Scope { implicit in =>
       val env     = assembly.map(defn => defn.name -> defn).toMap
       val workdir = VirtualDirectory.real(config.workdir)
 
       // Partition into multiple LLVM IR files proportional to number
       // of available processesors. This prevents LLVM from optimizing
       // across IR module boundary unless LTO is turned on.
-      def separate(): Unit =
-        partitionBy(assembly, procs)(_.name.top.mangle).par.foreach {
-          case (id, defns) =>
-            val sorted = defns.sortBy(_.name.show)
-            new Impl(config.targetTriple, env, sorted)
-              .gen(id.toString, workdir)
-        }
+      def separate(): Seq[Path] =
+        partitionBy(assembly, procs)(_.name.top.mangle).par
+          .map {
+            case (id, defns) =>
+              val sorted = defns.sortBy(_.name.show)
+              new Impl(config.targetTriple, env, sorted)
+                .gen(id.toString, workdir)
+          }
+          .toSeq
+          .seq
 
       // Generate a single LLVM IR file for the whole application.
       // This is an adhoc form of LTO. We use it in release mode if
       // Clang's LTO is not available.
-      def single(): Unit = {
+      def single(): Seq[Path] = {
         val sorted = assembly.sortBy(_.name.show)
-        new Impl(config.targetTriple, env, sorted)
-          .gen(id = "out", workdir)
+        Seq(
+          new Impl(config.targetTriple, env, sorted)
+            .gen(id = "out", workdir))
```

Comment:
I feel that add `targetTripple` for a name isn't bad idea here. Am I wrong?

Reply:
The `nativeTarget` in `sbt` gets set to `targetTriple` in the config. I think that is fine.

Reply:
The sbt naming and use has been cleared up below - `nativeTarget` is not needed at all.